scipy/stats/_continuous_distns.py

```diff
@@ -3542,10 +3542,16 @@ def _pdf(self, r):
         return r * exp(-0.5 * r**2)
 
     def _cdf(self, r):
-        return 1 - exp(-0.5 * r**2)
+        return -special.expm1(-r * r / 2.0)
```

Comment:
I agree with using `expm1` here, but not the rest of the change.

* `-r` allocates an array;
* `r / 2.0` allocates an array;
* multiplying these allocates an array.

By contrast, `-0.5 * r ** 2` allocates only two arrays. Also, division has to check for a zero second argument, so multiplication is to be preferred if the divisor is constant.